lib/common/ipc_client.c

```diff
@@ -1407,13 +1407,29 @@ pcmk__ipc_is_authentic_process_active(const char *name, uid_t refuid,
     int32_t qb_rc;
     pid_t found_pid = 0; uid_t found_uid = 0; gid_t found_gid = 0;
     qb_ipcc_connection_t *c;
+    struct pollfd pollfd = { 0, };
+    int poll_rc;
 
-    c = qb_ipcc_connect(name, 0);
+    c = qb_ipcc_connect_async(name, 0,
+                              &(pollfd.fd));
     if (c == NULL) {
         crm_info("Could not connect to %s IPC: %s", name, strerror(errno));
         rc = pcmk_rc_ipc_unresponsive;
         goto bail;
     }
+    pollfd.events = POLLIN;
+    do {
+        poll_rc = poll(&pollfd, 1, 2000);
```

Comment:
How did you pick the timeout value? I'm not really sure what the criteria should be ...

Reply:
big science - default-watchdog-timeout / 2 - a bit ;-)
no idea - open for discussion ...
thought doing multiple retries with a shorter timeout would be better than one with a large.
and implementation-wise it is as well easier as otherwise I would have to move it to being
really async. of course I can do that if we think we really need a shorter timeout.

Reply:
I'm fine with this approach. It might be worthwhile to do the pure async approach at some point.